src/main/scala/org/broadinstitute/dsde/workbench/sam/directory/DirectoryDAO.scala

```diff
@@ -19,18 +19,18 @@ trait DirectoryDAO {
 
   def loadSubjectFromEmail(email: String): Future[Option[WorkbenchSubject]]
 
-  def createUser(user: WorkbenchUser): Future[WorkbenchUser]
-  def loadUser(userId: WorkbenchUserId): Future[Option[WorkbenchUser]]
-  def loadUsers(userIds: Set[WorkbenchUserId]): Future[Seq[WorkbenchUser]]
-  def deleteUser(userId: WorkbenchUserId): Future[Unit]
+  def createUser(user: WorkbenchPerson): Future[WorkbenchPerson]
```

Comment:
I changed some method signatures to take the more generic `WorkbenchPerson`, which is a supertrait of `WorkbenchUser` and `WorkbenchUserServiceAccount`. See https://github.com/broadinstitute/workbench-libs/pull/33

Reply:
I am not sure I like the switch from WorkbenchUser to WorkbenchPerson (or Identity). I know I don't like the switch from WorkbenchUserId to WorkbenchSubject because WorkbenchSubject also includes WorkbenchGroupName which makes no sense here.

The only user related functions that make sense in terms of pets are create, load and delete. The calling code knows and cares which one it is doing so I think there should be different apis. I am ok with sharing code within the implementation.

The functions related to enable and list should remain only in terms of Users.

Reply:
OK, I can try going the route of having different APIs for users and pets (with maybe a shared private implementation). The `WorkbenchPerson` supertrait might not even be needed in that case. I was actually doing that originally but became annoyed at the code duplication in `DirectoryDAO`. But now I'm annoyed at this weird case class hierarchy. :)